bin/propolis-server/src/lib/initializer.rs

```diff
@@ -538,11 +538,29 @@ impl MachineInitializer<'_> {
                     );
                 }
 
-                let nworkers = NonZeroUsize::new(8).unwrap();
+                let nworkers: NonZeroUsize = match spec.workers {
+                    Some(workers) => {
+                        if (1..=propolis::block::MAX_FILE_WORKERS)
+                            .contains(&workers)
+                        {
+                            NonZeroUsize::new(workers).unwrap()
+                        } else {
+                            slog::warn!(
+                                self.log,
+                                "workers must be between 1 and {} \
+                                    Using default value of 8.",
+                                propolis::block::MAX_FILE_WORKERS
+                            );
+                            NonZeroUsize::new(8).unwrap()
```

Comment:
nit: probably define a const somewhere in here with the default worker count, rather than repeating the "magic" `8` a few times here?

Reply:
I've updated it to have a const